htlcswitch/test_utils.go

```diff
@@ -1045,6 +1053,34 @@ func newThreeHopNetwork(t testing.TB, aliceChannel, firstBobChannel,
 	}
 }
 
+// serverOption is a function which alters the three servers created for
+// a three hop network to allow custom settings on each server.
+type serverOption func(aliceServer, bobServer, carolServer *mockServer)
+
+// serverOptionWithHtlcNotifier is a functional option for the creation of
+// three hop network servers which allows setting of htlc notifiers.
+// Note that these notifiers should be started and stopped by the calling
+// function.
+func serverOptionWithHtlcNotifier(alice, bob,
+	carol *HtlcNotifier) serverOption {
+
+	return func(aliceServer, bobServer, carolServer *mockServer) {
+		aliceServer.htlcSwitch.cfg.HtlcNotifier = alice
```

Comment:
could we add the htlc notifier by default when creating the mockServers? It should not interfere with existing test cases,

Reply:
This gets a bit more involved, because we then need to pass mocked time into `newThreeHopNetwork` (which would probably need to be with functional options like this anyway, so all the other calls don't need to be updated), thread the notifier down to `initSwithcWithDB` (more functional opts so we don't need to update every caller) and either expand the `htlcNotifier` interface to include `Start`/`Stop`/`SubscribeEvents` or do a type assertion in the test to get the event subscription. If it's not a blocker, I think I'd rather leave it like this?